tests/dtm/test_single_process_restart.py

```diff
@@ -347,3 +356,80 @@ def test_bkt_creation_ios_after_m0d_restart(self):
+        s3_test_obj = S3TestLib(access_key=access_key, secret_key=secret_key,
+                                endpoint_url=S3_CFG["s3_url"])
+        self.log.info("Successfully created IAM user with name %s", self.s3acc_name)
+        self.iam_user = {'s3_acc': {'accesskey': access_key, 'secretkey': secret_key,
+                                    'user_name': self.s3acc_name}}
+
+        resp = self.ha_obj.create_bucket_to_complete_mpu(s3_data=self.iam_user,
+                                                         bucket_name=self.bucket_name,
+                                                         object_name=self.object_name,
+                                                         file_size=file_size,
+                                                         total_parts=total_parts,
+                                                         multipart_obj_path=multipart_obj_path)
+        assert_utils.assert_true(resp[0], resp)
+        result = s3_test_obj.object_info(self.bucket_name, self.object_name)
+        obj_size = result[1]["ContentLength"]
+        self.log.debug("Uploaded object info for %s is %s", self.bucket_name, result)
+        assert_utils.assert_equal(obj_size, file_size * const.Sizes.MB)
+        upload_checksum = str(resp[2])
+        self.log.info("Step 1: Successfully performed multipart upload for size 5GB.")
+
+        self.log.info("Step 2: Download the uploaded object and verify checksum")
+        resp = s3_test_obj.object_download(self.bucket_name, self.object_name, download_path)
+        self.log.info("Download object response: %s", resp)
+        assert_utils.assert_true(resp[0], resp[1])
+        download_checksum = self.ha_obj.cal_compare_checksum(file_list=[download_path],
+                                                             compare=False)[0]
+        assert_utils.assert_equal(upload_checksum, download_checksum,
+                                  f"Failed to match checksum: {upload_checksum},"
+                                  f" {download_checksum}")
+        self.log.info("Matched checksum: %s, %s", upload_checksum, download_checksum)
+        self.log.info("Step 2: Successfully downloaded the object and verified the checksum")
+
+        self.log.info("Step 3: Perform Single m0d Process Restart")
+        resp = self.dtm_obj.process_restart(master_node=self.master_node_list[0],
+                                            health_obj=self.health_obj,
+                                            pod_prefix=POD_NAME_PREFIX,
+                                            container_prefix=MOTR_CONTAINER_PREFIX,
+                                            process=self.m0d_process, check_proc_state=True)
+        assert_utils.assert_true(resp, "Failure in observed during process restart/recovery")
+        self.log.info("Step 3: m0d restarted and recovered successfully")
+
+        self.log.info("Step 4: Download the uploaded object after m0d recovery and verify checksum")
```

Comment:
so the purpose of the test to download uploaded object and verify checksum and not uploading another object? @ApurwaMohite  @priyankaborawake

Reply:
Yes, the intention is multiparts uploaded before m0d restart and consistent post recovery also.
@priyankaborawake please confirm once.

Reply:
@ApurwaMohite @sonalk0209  Yes, this test was scoped to check if written multipart object can be read post restart. It was conveyed multipart operation are not supported during/post restart.